### PR TITLE
Chart/sikri

### PR DESCRIPTION
**Pull request by mmedalen**

### CODE REVIEW

File: charts/fint-adapter-sikri/Chart.yaml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+# This is the chart version. This version number should be incremented each time you make changes
+# to the chart and its templates, including the app version.
+# Versions are expected to follow Semantic Versioning (https://semver.org/)
+version: 1.0.0
+
+# This is the version number of the application being deployed. This version number should be
+# incremented each time you make changes to the application. Versions are not expected to
+# follow Semantic Versioning. They should reflect the version the application is using.
+# It is recommended to use it with quotes.
+#appVersion: "3.9.0"
```
Comment (fsjovatsen) [inline charts/fint-adapter-sikri/Chart.yaml]: For application using the FINT model, the app version should be the FINT version. You can find this in the `apiVersion` in the `gradle.properties` file in the project. E.g. https://github.com/FINTLabs/fint-sikri-arkiv-adapter/blob/f283d0cc8300d182f45a51c40023f58e2d5d9d94/gradle.properties#L2

File: charts/fint-adapter-sikri/templates/deployment.yaml [hunk 2]
```diff
@@ -0,0 +1,38 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ .Values.nameOverride }}
```
Comment (fsjovatsen) [inline charts/fint-adapter-sikri/templates/deployment.yaml]: use `{{ include "fint-adapter-sikri.name" . }}` instead.

File: charts/fint-adapter-sikri/templates/deployment.yaml [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+      app: {{ .Values.nameOverride }}
+  template:
+    metadata:
+    spec:
+      volumes:
+        - name: tempdir
+          emptyDir:
+            sizeLimit: 1G
+      containers:
+        - name: {{ .Values.nameOverride }}
```
Comment (fsjovatsen) [inline charts/fint-adapter-sikri/templates/deployment.yaml]: use {{ include "fint-adapter-sikri.name" . }} instead.

File: charts/fint-adapter-sikri/templates/deployment.yaml [hunk 4]
```diff
@@ -0,0 +1,38 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ .Values.nameOverride }}
+spec:
+  replicas: {{ .Values.replicaCount }}
+  selector:
+    matchLabels:
+      app: {{ .Values.nameOverride }}
```
Comment (fsjovatsen) [inline charts/fint-adapter-sikri/templates/deployment.yaml]: use {{ include "fint-adapter-sikri.name" . }} instead.

File: charts/fint-adapter-sikri/templates/deployment.yaml [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+        - name: tempdir
+          emptyDir:
+            sizeLimit: 1G
+      containers:
+        - name: {{ .Values.nameOverride }}
+          image: "{{ .Values.image.repository }}"
+          #imagePullPolicy: {{ .Values.image.pullPolicy }}
+          ports:
+            - name: http
+              containerPort: 8091
```
Comment (fsjovatsen) [inline charts/fint-adapter-sikri/templates/deployment.yaml]: Should be a value

File: charts/fint-adapter-sikri/templates/deployment.yaml [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+              protocol: TCP
+          livenessProbe:
+            httpGet:
+              path: /
```
Comment (fsjovatsen) [inline charts/fint-adapter-sikri/templates/deployment.yaml:27]: use value

File: charts/fint-adapter-sikri/templates/deployment.yaml [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+              port: http
+          readinessProbe:
+            httpGet:
+              path: /
```
Comment (fsjovatsen) [inline charts/fint-adapter-sikri/templates/deployment.yaml:31]: use value

File: charts/fint-adapter-sikri/templates/deployment.yaml [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+              port: http
+          resources:
+            {{- toYaml .Values.resources | nindent 12 }}
```
Comment (fsjovatsen) [inline charts/fint-adapter-sikri/templates/deployment.yaml:34]: Do something like this:

```
         resources:
            limits:
              memory: {{ .Values.consumer.deployment.resources.limitsMemory }}
              cpu: {{ .Values.consumer.deployment.resources.limitsCpu }}
            requests:
              memory: {{ .Values.consumer.deployment.resources.requestsMemory }}
              cpu: {{ .Values.consumer.deployment.resources.requestsCpu }}
```

Ref. https://helm.sh/docs/chart_best_practices/values/#flat-or-nested-values

File: charts/fint-adapter-sikri/templates/fint-adapter-op-item.yaml [hunk 9]
```diff
@@ -0,0 +1,9 @@
+{{- if .Values.onePasswordVault.create -}}
+apiVersion: onepassword.com/v1
+kind: OnePasswordItem
+metadata:
+  name: {{ .Values.nameOverride }}
```
Comment (fsjovatsen) [inline charts/fint-adapter-sikri/templates/fint-adapter-op-item.yaml]: use `{{ include "fint-adapter-sikri.name" . }}` instead.

maybe it should be suffixed with `-secret`?

File: charts/fint-adapter-sikri/values-innlandet.yaml [hunk 10]
```diff
@@ -0,0 +1,157 @@
+# Default values for fint-adapter-sikri.
+# This is a YAML-formatted file.
+# Declare variables to be passed into your templates.
+
+replicaCount: 1
+
+nameOverride: adapter-sikri-arkiv-innlandetfylke
+
+onePasswordVault:
```
Comment (fsjovatsen) [inline charts/fint-adapter-sikri/values-innlandet.yaml]: This is not a vault but an item. It should probably be called `onePasswordItem`.

File: charts/fint-adapter-sikri/values-innlandet.yaml [hunk 11]
```diff
@@ -0,0 +1,157 @@
...
+volumes:
+  - name: tempdir
+    emptyDir:
+      sizeLimit: 1G
+volumeMounts:
+  - name: tempdir
+    mountPath: /temp
+
+image:
+  repository: fintlabsacr.azurecr.io/sikri-arkiv-adapter:PR-38.37
```
Comment (fsjovatsen) [inline charts/fint-adapter-sikri/values-innlandet.yaml]: Tag should be removed. Tag is a separate value.

File: charts/fint-adapter-sikri/templates/fint-adapter-op-item.yaml [hunk 12]
```diff
@@ -0,0 +1,9 @@
+{{- if .Values.onePasswordVault.create -}}
```
Comment (fsjovatsen) [inline charts/fint-adapter-sikri/templates/fint-adapter-op-item.yaml]: There should be one OnePasswordItem pr. secret

File: charts/fint-adapter-sikri/values-innlandet.yaml [hunk 13]
```diff
@@ -0,0 +1,157 @@
...
+  annotations: {}
+  # The name of the service account to use.
+  # If not set and create is true, a name is generated using the fullname template
+  name: ""
+
+service:
+  type: ClusterIP
+  port: 80
+
+secretRef:
```
Comment (fsjovatsen) [inline charts/fint-adapter-sikri/values-innlandet.yaml:46]: Move to deployment

**Review by fsjovatsen — DISMISSED** [review]
@erlingjahr will walk you through the PR.

File: charts/fint-adapter-sikri/templates/fint-adapter-op-item.yaml (hunk 12 above)
Comment (mmedalen) [reply charts/fint-adapter-sikri/templates/fint-adapter-op-item.yaml]: could you elaborate on that?

**Pull request merged**